app/questionnaire/router.py

```diff
@@ -214,15 +210,21 @@ def get_section_return_location_when_section_complete(
         self, routing_path
     ) -> Location:
 
-        section_summary_location = self._get_location_of_section_summary(routing_path)
-        if section_summary_location:
-            return section_summary_location
+        last_block_id = routing_path[-1]
+        block = self._schema.get_block(last_block_id)
```

Comment:
Minor, but this would be slightly clearer when used below if it was named `last_block`.